laurent22/joplin
# Mobile: Fixes #10252 Not able to change notebook when using 'New Note' quick action

Issue https://github.com/laurent22/joplin/issues/10252

# Summary

Quick actions are shortcuts that the user can use to access functionalities of the app outside of it.
On Android, by long pressing the app we have two quick actions 'New Note' and 'New to-do'

The issue would happen when using the quick action after the app was already opened and the selected folder was a different one from when the app was open. Because of how the code was implemented the selected folder would always be the same first one since the `selectedFolderId` would never get refreshed.

I refactored the implementation to fix that.

# Testing

1. Create two notebooks
2. Open Joplin with 'New Note' quick action
3. Change notebook with notebook picker (from note header)
4. Finish creating the note, it should be inside the last selected notebook
5. Move Joplin to the background (not closing)
6. Use quick action again
7. Changing notebook should still be possible
8. The new note should be created in the last selected folder
